### DISCUSSION TITLE
ConvTasNet Saving / Loading Model

### DISCUSSION THREAD

**Discussion by cemox35** [asteroid-team/asteroid#563] [Q&A] [answered]
Hi,

I use the ConvTasNet model in Asteroid and Trainer automatically saves the model's checkpoints in PyTorch-lightning file, however, I cannot use **_"load_from_checkpoint"_** or _**"from_pretrained".**_  

What is the correct solution of the saving and loading models use for the test, not training?

**Comment by mpariente** [comment]
Can you show how you do the loading and the error message?

**Reply by cemox35** [reply]
First try:

    ENCODER = ["stft", "stft", "stft", "free", "free", "free"]
    MODEL_PATH = ["lightning_logs/version_53/checkpoints/epoch=4-step=1444.ckpt"]  # write model path
    i = 0

    N = [512, 512, 512, 256]
    L = [40, 40, 16, ]  # 2.5ms
    B = [128, 128, 128, 256]
    H = [256, 512, 512, 512]
    Sc = [128, 512, 128, ]  # ??
    P = [3, 3, 3, 3]
    X = [7, 7, 8, 8]
    R = [2, 2, 3, 4]

    SAMPLE_RATE = 8000

    model = ConvTasNet(
        n_src=SPEAKER,
        n_filters=N[i % 3],
        kernel_size=L[i % 3],
        bn_chan=B[i % 3],
        skip_chan=Sc[i % 3],
        hid_chan=H[i % 3],
        conv_kernel_size=P[i % 3],
        n_blocks=X[i % 3],
        n_repeats=R[i % 3],
        sample_rate=SAMPLE_RATE,
        fb_name=ENCODER[i]
    ).double()

    model.from_pretrained(pretrained_model_conf_or_path=MODEL_PATH[i])`

ValueError: Expected config dictionary to have field model_name`. Found only: dict_keys(['epoch', 'global_step', 'pytorch-lightning_version', 'state_dict', 'callbacks', 'optimizer_states', 'lr_schedulers', 'native_amp_scaling_state', 'training_config'])

When I tried to "load_from_checkpoint":

`model = ConvTasNet.load_from_checkpoint(MODEL_PATH[i])`

AttributeError: type object 'ConvTasNet' has no attribute 'load_from_checkpoint'

Last try:

`model = model.load_state_dict(torch.load(MODEL_PATH[i]))`

RuntimeError: Error(s) in loading state_dict for ConvTasNet:
	Missing key(s) in state_dict: "encoder.filterbank._filters", "masker.bottleneck.0.gamma", "masker.bottleneck.0.beta", "masker.bottleneck.1.weight", "masker.bottleneck.1.bias", "masker.TCN.0.shared_block.0.weight", "masker.TCN.0.shared_block.0.bias", "masker.TCN.0.shared_block.1.weight", "masker.TCN.0.shared_block.2.gamma", "masker.TCN.0.shared_block.2.beta", "masker.TCN.0.shared_block.3.weight", "masker.TCN.0.shared_block.3.bias", "masker.TCN.0.shared_block.4.weight", "masker.TCN.0.shared_block.5.gamma", "masker.TCN.0.shared_block.5.beta", "masker.TCN.0.res_conv.weight", "masker.TCN.0.res_conv.bias", "masker.TCN.0.skip_conv.weight", "masker.TCN.0.skip_conv.bias", "masker.TCN.1.shared_block.0.weight", "masker.TCN.1.shared_block.0.bias", "masker.TCN.1.shared_block.1.weight", "masker.TCN.1.shared_block.2.gamma", "masker.TCN.1.shared_block.2.beta", "masker.TCN.1.shared_block.3.weight", "masker.TCN.1.shared_block.3.bias", "masker.TCN.1.shared_block.4.weight", "masker.TCN.1.shared_block.5.gamma", "masker.TCN.1.shared_block.5.beta", "masker.TCN.1.res_conv.weight", "masker.TCN.1.res_conv.bias", "masker.TCN.1.skip_conv.weight", "masker.TCN.1.skip_conv.bias", "masker.TCN.2.shared_block.0.weight", "masker.TCN.2.shared_block.0.bias", "masker.TCN.2.shared_block.1.weight", "masker.TCN.2.shared_block.2.gamma", "masker.TCN.2.shared_block.2.beta", "masker.TCN.2.shared_block.3.weight", "masker.TCN.2.shared_block.3.bias", "masker.TCN.2.shared_block.4.weight", "masker.TCN.2.shared_block.5.gamma", "masker.TCN.2.shared_block.5.beta", "masker.TCN.2.res_conv.weight", "masker.TCN.2.res_conv.bias", "masker.TCN.2.skip_conv.weight", "masker.TCN.2.skip_conv.bias", "masker.TCN.3.shared_block.0.weight", "masker.TCN.3.shared_block.0.bias", "masker.TCN.3.shared_block.1.weight", "masker.TCN.3.shared_block.2.gamma", "masker.TCN.3.shared_block.2.beta", "masker.TCN.3.shared_block.3.weight", "masker.TCN.3.shared_block.3.bias", "masker.TCN.3.shared_block.4.weight", "masker.TCN.3.shared_block.5.gamma", "masker.TCN.3.shared_block.5.beta", "masker.TCN.3.res_conv.weight", "masker.TCN.3.res_conv.bias", "masker.TCN.3.skip_conv.weight", "masker.TCN.3.skip_conv.bias", "masker.TCN.4.shared_block.0.weight", "masker.TCN.4.shared_block.0.bias", "masker.TCN.4.shared_block.1.weight", "masker.TCN.4.shared_block.2.gamma", "masker.TCN.4.shared_block.2.beta", "masker.TCN.4.shared_block.3.weight", "masker.TCN.4.shared_block.3.bias", "masker.TCN.4.shared_block.4.weight", "masker.TCN.4.shared_block.5.gamma", "masker.TCN.4.shared_block.5.beta", "masker.TCN.4.res_conv.weight", "masker.TCN.4.res_conv.bias", "masker.TCN.4.skip_conv.weight", "masker.TCN.4.skip_conv.bias", "masker.TCN.5.shared_block.0.weight", "masker.TCN.5.shared_block.0.bias", "masker.TCN.5.shared_block.1.weight", "masker.TCN.5.shared_block.2.gamma", "masker.TCN.5.shared_block.2.beta", "masker.TCN.5.shared_block.3.weight", "masker.TCN.5.shared_block.3.bias", "masker.TCN.5.shared_block.4.weight", "masker.TCN.5.shared_block.5.gamma", "masker.TCN.5.shared_block.5.beta", "masker.TCN.5.res_conv.weight", "masker.TCN.5.res_conv.bias", "masker.TCN.5.skip_conv.weight", "masker.TCN.5.skip_conv.bias", "masker.TCN.6.shared_block.0.weight", "masker.TCN.6.shared_block.0.bias", "masker.TCN.6.shared_block.1.weight", "masker.TCN.6.shared_block.2.gamma", "masker.TCN.6.shared_block.2.beta", "masker.TCN.6.shared_block.3.weight", "masker.TCN.6.shared_block.3.bias", "masker.TCN.6.shared_block.4.weight", "masker.TCN.6.shared_block.5.gamma", "masker.TCN.6.shared_block.5.beta", "masker.TCN.6.res_conv.weight", "masker.TCN.6.res_conv.bias", "masker.TCN.6.skip_conv.weight", "masker.TCN.6.skip_conv.bias", "masker.TCN.7.shared_block.0.weight", "masker.TCN.7.shared_block.0.bias", "masker.TCN.7.shared_block.1.weight", "masker.TCN.7.shared_block.2.gamma", "masker.TCN.7.shared_block.2.beta", "masker.TCN.7.shared_block.3.weight", "masker.TCN.7.shared_block.3.bias", "masker.TCN.7.shared_block.4.weight", "masker.TCN.7.shared_block.5.gamma", "masker.TCN.7.shared_block.5.beta", "masker.TCN.7.res_conv.weight", "masker.TCN.7.res_conv.bias", "masker.TCN.7.skip_conv.weight", "masker.TCN.7.skip_conv.bias", "masker.TCN.8.shared_block.0.weight", "masker.TCN.8.shared_block.0.bias", "masker.TCN.8.shared_block.1.weight", "masker.TCN.8.shared_block.2.gamma", "masker.TCN.8.shared_block.2.beta", "masker.TCN.8.shared_block.3.weight", "masker.TCN.8.shared_block.3.bias", "masker.TCN.8.shared_block.4.weight", "masker.TCN.8.shared_block.5.gamma", "masker.TCN.8.shared_block.5.beta", "masker.TCN.8.res_conv.weight", "masker.TCN.8.res_conv.bias", "masker.TCN.8.skip_conv.weight", "masker.TCN.8.skip_conv.bias", "masker.TCN.9.shared_block.0.weight", "masker.TCN.9.shared_block.0.bias", "masker.TCN.9.shared_block.1.weight", "masker.TCN.9.shared_block.2.gamma", "masker.TCN.9.shared_block.2.beta", "masker.TCN.9.shared_block.3.weight", "masker.TCN.9.shared_block.3.bias", "masker.TCN.9.shared_block.4.weight", "masker.TCN.9.shared_block.5.gamma", "masker.TCN.9.shared_block.5.beta", "masker.TCN.9.res_conv.weight", "masker.TCN.9.res_conv.bias", "masker.TCN.9.skip_conv.weight", "masker.TCN.9.skip_conv.bias", "masker.TCN.10.shared_block.0.weight", "masker.TCN.10.shared_block.0.bias", "masker.TCN.10.shared_block.1.weight", "masker.TCN.10.shared_block.2.gamma", "masker.TCN.10.shared_block.2.beta", "masker.TCN.10.shared_block.3.weight", "masker.TCN.10.shared_block.3.bias", "masker.TCN.10.shared_block.4.weight", "masker.TCN.10.shared_block.5.gamma", "masker.TCN.10.shared_block.5.beta", "masker.TCN.10.res_conv.weight", "masker.TCN.10.res_conv.bias", "masker.TCN.10.skip_conv.weight", "masker.TCN.10.skip_conv.bias", "masker.TCN.11.shared_block.0.weight", "masker.TCN.11.shared_block.0.bias", "masker.TCN.11.shared_block.1.weight", "masker.TCN.11.shared_block.2.gamma", "masker.TCN.11.shared_block.2.beta", "masker.TCN.11.shared_block.3.weight", "masker.TCN.11.shared_block.3.bias", "masker.TCN.11.shared_block.4.weight", "masker.TCN.11.shared_block.5.gamma", "masker.TCN.11.shared_block.5.beta", "masker.TCN.11.res_conv.weight", "masker.TCN.11.res_conv.bias", "masker.TCN.11.skip_conv.weight", "masker.TCN.11.skip_conv.bias", "masker.TCN.12.shared_block.0.weight", "masker.TCN.12.shared_block.0.bias", "masker.TCN.12.shared_block.1.weight", "masker.TCN.12.shared_block.2.gamma", "masker.TCN.12.shared_block.2.beta", "masker.TCN.12.shared_block.3.weight", "masker.TCN.12.shared_block.3.bias", "masker.TCN.12.shared_block.4.weight", "masker.TCN.12.shared_block.5.gamma", "masker.TCN.12.shared_block.5.beta", "masker.TCN.12.res_conv.weight", "masker.TCN.12.res_conv.bias", "masker.TCN.12.skip_conv.weight", "masker.TCN.12.skip_conv.bias", "masker.TCN.13.shared_block.0.weight", "masker.TCN.13.shared_block.0.bias", "masker.TCN.13.shared_block.1.weight", "masker.TCN.13.shared_block.2.gamma", "masker.TCN.13.shared_block.2.beta", "masker.TCN.13.shared_block.3.weight", "masker.TCN.13.shared_block.3.bias", "masker.TCN.13.shared_block.4.weight", "masker.TCN.13.shared_block.5.gamma", "masker.TCN.13.shared_block.5.beta", "masker.TCN.13.res_conv.weight", "masker.TCN.13.res_conv.bias", "masker.TCN.13.skip_conv.weight", "masker.TCN.13.skip_conv.bias", "masker.mask_net.0.weight", "masker.mask_net.1.weight", "masker.mask_net.1.bias", "decoder.filterbank._filters". 
	Unexpected key(s) in state_dict: "epoch", "global_step", "pytorch-lightning_version", "state_dict", "callbacks", "optimizer_states", "lr_schedulers", "native_amp_scaling_state", "training_config".

**Reply by mpariente** [reply]
You should probably load the state_dict only. 

Can you try to reproduce the problem with a minimal script, I'll help from there.

**Reply by cemox35** [reply]
This is my test code:

import torch
from asteroid.models import ConvTasNet

def test_models():
  

    MODEL_PATH = ["lightning_logs/version_53/checkpoints/epoch=4-step=1444.ckpt"]  # write model path

    i = 0
    SPEAKER = 1

    ENCODER = ["stft", "stft", "stft", "free", "free", "free"]

    N = [512, 512, 512, 256]
    L = [40, 40, 16, ]  # 2.5ms
    B = [128, 128, 128, 256]
    H = [256, 512, 512, 512]
    Sc = [128, 512, 128, ]  # ??
    P = [3, 3, 3, 3]
    X = [7, 7, 8, 8]
    R = [2, 2, 3, 4]

    SAMPLE_RATE = 8000

    model = ConvTasNet(
        n_src=SPEAKER,
        n_filters=N[i % 3],
        kernel_size=L[i % 3],
        bn_chan=B[i % 3],
        skip_chan=Sc[i % 3],
        hid_chan=H[i % 3],
        conv_kernel_size=P[i % 3],
        n_blocks=X[i % 3],
        n_repeats=R[i % 3],
        sample_rate=SAMPLE_RATE,
        fb_name=ENCODER[i]
    ).double()

    model = model.load_state_dict(torch.load(MODEL_PATH[i]))

    model.separate("s1.wav", resample=True)


if __name__ == '__main__':
    test_models()
`

Its give that error: 

Traceback (most recent call last):
  File "F:\ALL PROJECTS\Pycharm Projects\Thesis\train.py", line 179, in <module>
    test_models()
  File "F:\ALL PROJECTS\Pycharm Projects\Thesis\train.py", line 170, in test_models
    model = model.load_state_dict(torch.load(MODEL_PATH[i]))
  File "C:\conda\envs\thesis\lib\site-packages\torch\nn\modules\module.py", line 1406, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for ConvTasNet:
	Missing key(s) in state_dict: "encoder.filterbank._filters", "masker.bottleneck.0.gamma", "masker.bottleneck.0.beta", "masker.bottleneck.1.weight", "masker.bottleneck.1.bias", "masker.TCN.0.shared_block.0.weight", "masker.TCN.0.shared_block.0.bias", "masker.TCN.0.shared_block.1.weight", "masker.TCN.0.shared_block.2.gamma", "masker.TCN.0.shared_block.2.beta", "masker.TCN.0.shared_block.3.weight", "masker.TCN.0.shared_block.3.bias", "masker.TCN.0.shared_block.4.weight", "masker.TCN.0.shared_block.5.gamma", "masker.TCN.0.shared_block.5.beta", "masker.TCN.0.res_conv.weight", "masker.TCN.0.res_conv.bias", "masker.TCN.0.skip_conv.weight", "masker.TCN.0.skip_conv.bias", "masker.TCN.1.shared_block.0.weight", "masker.TCN.1.shared_block.0.bias", "masker.TCN.1.shared_block.1.weight", "masker.TCN.1.shared_block.2.gamma", "masker.TCN.1.shared_block.2.beta", "masker.TCN.1.shared_block.3.weight", "masker.TCN.1.shared_block.3.bias", "masker.TCN.1.shared_block.4.weight", "masker.TCN.1.shared_block.5.gamma", "masker.TCN.1.shared_block.5.beta", "masker.TCN.1.res_conv.weight", "masker.TCN.1.res_conv.bias", "masker.TCN.1.skip_conv.weight", "masker.TCN.1.skip_conv.bias", "masker.TCN.2.shared_block.0.weight", "masker.TCN.2.shared_block.0.bias", "masker.TCN.2.shared_block.1.weight", "masker.TCN.2.shared_block.2.gamma", "masker.TCN.2.shared_block.2.beta", "masker.TCN.2.shared_block.3.weight", "masker.TCN.2.shared_block.3.bias", "masker.TCN.2.shared_block.4.weight", "masker.TCN.2.shared_block.5.gamma", "masker.TCN.2.shared_block.5.beta", "masker.TCN.2.res_conv.weight", "masker.TCN.2.res_conv.bias", "masker.TCN.2.skip_conv.weight", "masker.TCN.2.skip_conv.bias", "masker.TCN.3.shared_block.0.weight", "masker.TCN.3.shared_block.0.bias", "masker.TCN.3.shared_block.1.weight", "masker.TCN.3.shared_block.2.gamma", "masker.TCN.3.shared_block.2.beta", "masker.TCN.3.shared_block.3.weight", "masker.TCN.3.shared_block.3.bias", "masker.TCN.3.shared_block.4.weight", "masker.TCN.3.shared_block.5.gamma", "masker.TCN.3.shared_block.5.beta", "masker.TCN.3.res_conv.weight", "masker.TCN.3.res_conv.bias", "masker.TCN.3.skip_conv.weight", "masker.TCN.3.skip_conv.bias", "masker.TCN.4.shared_block.0.weight", "masker.TCN.4.shared_block.0.bias", "masker.TCN.4.shared_block.1.weight", "masker.TCN.4.shared_block.2.gamma", "masker.TCN.4.shared_block.2.beta", "masker.TCN.4.shared_block.3.weight", "masker.TCN.4.shared_block.3.bias", "masker.TCN.4.shared_block.4.weight", "masker.TCN.4.shared_block.5.gamma", "masker.TCN.4.shared_block.5.beta", "masker.TCN.4.res_conv.weight", "masker.TCN.4.res_conv.bias", "masker.TCN.4.skip_conv.weight", "masker.TCN.4.skip_conv.bias", "masker.TCN.5.shared_block.0.weight", "masker.TCN.5.shared_block.0.bias", "masker.TCN.5.shared_block.1.weight", "masker.TCN.5.shared_block.2.gamma", "masker.TCN.5.shared_block.2.beta", "masker.TCN.5.shared_block.3.weight", "masker.TCN.5.shared_block.3.bias", "masker.TCN.5.shared_block.4.weight", "masker.TCN.5.shared_block.5.gamma", "masker.TCN.5.shared_block.5.beta", "masker.TCN.5.res_conv.weight", "masker.TCN.5.res_conv.bias", "masker.TCN.5.skip_conv.weight", "masker.TCN.5.skip_conv.bias", "masker.TCN.6.shared_block.0.weight", "masker.TCN.6.shared_block.0.bias", "masker.TCN.6.shared_block.1.weight", "masker.TCN.6.shared_block.2.gamma", "masker.TCN.6.shared_block.2.beta", "masker.TCN.6.shared_block.3.weight", "masker.TCN.6.shared_block.3.bias", "masker.TCN.6.shared_block.4.weight", "masker.TCN.6.shared_block.5.gamma", "masker.TCN.6.shared_block.5.beta", "masker.TCN.6.res_conv.weight", "masker.TCN.6.res_conv.bias", "masker.TCN.6.skip_conv.weight", "masker.TCN.6.skip_conv.bias", "masker.TCN.7.shared_block.0.weight", "masker.TCN.7.shared_block.0.bias", "masker.TCN.7.shared_block.1.weight", "masker.TCN.7.shared_block.2.gamma", "masker.TCN.7.shared_block.2.beta", "masker.TCN.7.shared_block.3.weight", "masker.TCN.7.shared_block.3.bias", "masker.TCN.7.shared_block.4.weight", "masker.TCN.7.shared_block.5.gamma", "masker.TCN.7.shared_block.5.beta", "masker.TCN.7.res_conv.weight", "masker.TCN.7.res_conv.bias", "masker.TCN.7.skip_conv.weight", "masker.TCN.7.skip_conv.bias", "masker.TCN.8.shared_block.0.weight", "masker.TCN.8.shared_block.0.bias", "masker.TCN.8.shared_block.1.weight", "masker.TCN.8.shared_block.2.gamma", "masker.TCN.8.shared_block.2.beta", "masker.TCN.8.shared_block.3.weight", "masker.TCN.8.shared_block.3.bias", "masker.TCN.8.shared_block.4.weight", "masker.TCN.8.shared_block.5.gamma", "masker.TCN.8.shared_block.5.beta", "masker.TCN.8.res_conv.weight", "masker.TCN.8.res_conv.bias", "masker.TCN.8.skip_conv.weight", "masker.TCN.8.skip_conv.bias", "masker.TCN.9.shared_block.0.weight", "masker.TCN.9.shared_block.0.bias", "masker.TCN.9.shared_block.1.weight", "masker.TCN.9.shared_block.2.gamma", "masker.TCN.9.shared_block.2.beta", "masker.TCN.9.shared_block.3.weight", "masker.TCN.9.shared_block.3.bias", "masker.TCN.9.shared_block.4.weight", "masker.TCN.9.shared_block.5.gamma", "masker.TCN.9.shared_block.5.beta", "masker.TCN.9.res_conv.weight", "masker.TCN.9.res_conv.bias", "masker.TCN.9.skip_conv.weight", "masker.TCN.9.skip_conv.bias", "masker.TCN.10.shared_block.0.weight", "masker.TCN.10.shared_block.0.bias", "masker.TCN.10.shared_block.1.weight", "masker.TCN.10.shared_block.2.gamma", "masker.TCN.10.shared_block.2.beta", "masker.TCN.10.shared_block.3.weight", "masker.TCN.10.shared_block.3.bias", "masker.TCN.10.shared_block.4.weight", "masker.TCN.10.shared_block.5.gamma", "masker.TCN.10.shared_block.5.beta", "masker.TCN.10.res_conv.weight", "masker.TCN.10.res_conv.bias", "masker.TCN.10.skip_conv.weight", "masker.TCN.10.skip_conv.bias", "masker.TCN.11.shared_block.0.weight", "masker.TCN.11.shared_block.0.bias", "masker.TCN.11.shared_block.1.weight", "masker.TCN.11.shared_block.2.gamma", "masker.TCN.11.shared_block.2.beta", "masker.TCN.11.shared_block.3.weight", "masker.TCN.11.shared_block.3.bias", "masker.TCN.11.shared_block.4.weight", "masker.TCN.11.shared_block.5.gamma", "masker.TCN.11.shared_block.5.beta", "masker.TCN.11.res_conv.weight", "masker.TCN.11.res_conv.bias", "masker.TCN.11.skip_conv.weight", "masker.TCN.11.skip_conv.bias", "masker.TCN.12.shared_block.0.weight", "masker.TCN.12.shared_block.0.bias", "masker.TCN.12.shared_block.1.weight", "masker.TCN.12.shared_block.2.gamma", "masker.TCN.12.shared_block.2.beta", "masker.TCN.12.shared_block.3.weight", "masker.TCN.12.shared_block.3.bias", "masker.TCN.12.shared_block.4.weight", "masker.TCN.12.shared_block.5.gamma", "masker.TCN.12.shared_block.5.beta", "masker.TCN.12.res_conv.weight", "masker.TCN.12.res_conv.bias", "masker.TCN.12.skip_conv.weight", "masker.TCN.12.skip_conv.bias", "masker.TCN.13.shared_block.0.weight", "masker.TCN.13.shared_block.0.bias", "masker.TCN.13.shared_block.1.weight", "masker.TCN.13.shared_block.2.gamma", "masker.TCN.13.shared_block.2.beta", "masker.TCN.13.shared_block.3.weight", "masker.TCN.13.shared_block.3.bias", "masker.TCN.13.shared_block.4.weight", "masker.TCN.13.shared_block.5.gamma", "masker.TCN.13.shared_block.5.beta", "masker.TCN.13.res_conv.weight", "masker.TCN.13.res_conv.bias", "masker.TCN.13.skip_conv.weight", "masker.TCN.13.skip_conv.bias", "masker.mask_net.0.weight", "masker.mask_net.1.weight", "masker.mask_net.1.bias", "decoder.filterbank._filters". 
	Unexpected key(s) in state_dict: "epoch", "global_step", "pytorch-lightning_version", "state_dict", "callbacks", "optimizer_states", "lr_schedulers", "native_amp_scaling_state", "training_config".

**Reply by mpariente** [reply]
I meant with the model saving part on the same code example. So that the example is standalone.

**Reply by cemox35** [reply]
I don't use any extra model saving code. Trainer is automatically logging these. Should I write my own model saving code? If yes, how can I do that?

**Comment by cemox35** [comment, marked as answer]
Oh, I think, I need a use `System` instead `ConvTasNet`. However, I got another error.

This is my code:
`MODEL_PATH = ["lightning_logs/version_53/checkpoints/epoch=4-step=1444.ckpt"]  # write model path

    i = 5

    train_loader, val_loader = load_data()

    model = ConvTasNet(
        n_src=SPEAKER,
        n_filters=N[i % 3],
        kernel_size=L[i % 3],
        bn_chan=B[i % 3],
        skip_chan=Sc[i % 3],
        hid_chan=H[i % 3],
        conv_kernel_size=P[i % 3],
        n_blocks=X[i % 3],
        n_repeats=R[i % 3],
        sample_rate=SAMPLE_RATE,
        fb_name=ENCODER[i]
    ).double()

    if torch.cuda.is_available():
        device = "cuda"
    else:
        device = "cpu"
    print(f"Using {device}")

    model.to(device, non_blocking=True)

    loss = PITLossWrapper(singlesrc_neg_sisdr, pit_from="pw_pt")
    # loss = PITLossWrapper(pairwise_neg_sisdr, pit_from="pt_mtx")

    optimizer = optim.Adam(model.parameters(), lr=1e-3)

    system = System(model, optimizer, loss, train_loader, val_loader)

    system = system.load_from_checkpoint(MODEL_PATH[0])

    print(system)`

the error is: `TypeError: __init__() missing 4 required positional arguments: 'model', 'optimizer', 'loss_func', and 'train_loader'`

**Reply by cemox35** [reply]
Well, I solved my problem with using `Trainer`.

Full example code: 

      MODEL_PATH = ["lightning_logs/version_53/checkpoints/epoch=4-step=1444.ckpt"] # write model path
      i = 5
      
      train_loader, val_loader = load_data()
      
      model = ConvTasNet(
        n_src=SPEAKER,
        n_filters=N[i % 3],
        kernel_size=L[i % 3],
        bn_chan=B[i % 3],
        skip_chan=Sc[i % 3],
        hid_chan=H[i % 3],
        conv_kernel_size=P[i % 3],
        n_blocks=X[i % 3],
        n_repeats=R[i % 3],
        sample_rate=SAMPLE_RATE,
        fb_name=ENCODER[i]
      ).double()
      
      if torch.cuda.is_available():
        device = "cuda"
      else:
        device = "cpu"
      print(f"Using {device}")
      
      model.to(device, non_blocking=True)
      
      loss = PITLossWrapper(singlesrc_neg_sisdr, pit_from="pw_pt")
      
      optimizer = optim.Adam(model.parameters(), lr=1e-3)
      
      system = System(model, optimizer, loss, train_loader, val_loader)
      
      trainer = Trainer(max_epochs=EPOCHS,
      resume_from_checkpoints=MODEL_PATH[0],
      gpus=1)
      
      trainer.fit(system)
      
      model.float().separate("s1.wav", resample=True)





After all, I can get the s1_est.wav file.
